src/python/grpcio/grpc/_channel.py

```diff
@@ -1026,6 +1016,7 @@ def __init__(self, target, options, credentials, compression):
           compression: An optional value indicating the compression method to be
             used over the lifetime of the channel.
         """
+        self._connectivity_state = None  # ensure attribute exists for __del__
```

Comment:
Is this line still needed with your try/catch in `__del__`?

Reply:
`try/catch` will catch this, but ideally we should move towards fixing the logic so exception never occurs and perhaps remove try/catch afterwards.

Reply:
In line 1024, if the `__init__` executed without error, the `_connectivity_state` will be set properly. Hence, I still think we should just keep one of them.

Reply:
What if the `__init__` is not executed properly? i.e. what if `cygrpc.Channel` constructor raises an exception?  We don't have the `_channel` instance to initialize `_connectivity_state` before calling that constructor of `cygrpc.Channel` so there's a chicken-and-egg problem here. That was the purpose of `hasattr` check as well: the cygrpc.Channel constructor potentially raising.

Reply:
I would say in the case of `cygrpc.Channel` constructor raises. Your update should catch the exception anyway. Alternatively, you can specify which kind of exception you want to silent in the `__del__` function.

Reply:
I see your point: my hope is to kill `try/except` in `__del__` at some point, so don't want to assume it will be there.

Reply:
I think remove this line is easier for you than the alternative solution of guarding all known conditions in `__del__`.